servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp

```diff
@@ -855,6 +857,41 @@ void fragment() {
 		overdraw_material_uniform_set = md->uniform_set;
 	}
 
+	{
+		uv2_texel_density_material_shader = material_storage->shader_allocate();
+		material_storage->shader_initialize(uv2_texel_density_material_shader);
+		material_storage->shader_set_code(uv2_texel_density_material_shader, R"(
+// 3D editor UV2 texel density debug draw mode shader (clustered).
+
+shader_type spatial;
+
+instance uniform vec2 lightmap_size : instance_index(0);
+
+void fragment() {
+	vec2 lightmap_pos = UV2 * lightmap_size;
+	vec2 p = lightmap_pos / 10.0;
+
+	vec2 ddx = dFdx(UV2);
+	vec2 ddy = dFdy(UV2);
+
+	vec2 w = max(abs(ddx), abs(ddy)) + 0.01;
+
+	vec2 s = 2.0*(abs(fract((p-0.5*w)/2.0)-0.5)-abs(fract((p+0.5*w)/2.0)-0.5))/w;
+	float t = .5 - .5*s.x*s.y;
```

Comment:
Some suggestions to comply with our [shader style guide](https://docs.godotengine.org/en/latest/tutorials/shaders/shaders_style_guide.html)
```suggestion
	vec2 s = 2.0 * (abs(fract((p - 0.5 * w) / 2.0) - 0.5) - abs(fract((p + 0.5 * w) / 2.0) - 0.5)) / w;
	float t = 0.5 - 0.5 * s.x * s.y;
```